src/cli/serve/serve.js

```diff
@@ -112,10 +112,17 @@ function applyConfigOverrides(rawConfig, opts, extraCliOptions) {
   if (opts.elasticsearch) set('elasticsearch.hosts', opts.elasticsearch.split(','));
   if (opts.port) set('server.port', opts.port);
   if (opts.host) set('server.host', opts.host);
-  if (opts.quiet) set('logging.quiet', true);
-  if (opts.silent) set('logging.silent', true);
-  if (opts.verbose) set('logging.verbose', true);
-  if (opts.logFile) set('logging.dest', opts.logFile);
+  if (opts.silent) {
+    set('logging.silent', true) && set('logging.root.level', 'off');
+  }
+  if (opts.verbose) {
+    // Only set logging.verbose to true when KP logging isn't configured.
+    if (!has('rawConfig.logging.root.appenders')) {
+      set('logging.verbose', true);
```

Comment:
I think it's `has('logging.root.appenders')` ?

Reply:
Not exactly. What we're doing here is setting the legacy logging to `verbose` only when we _don't_ have an appender configured in the `rawConfig` (read from the actual `kibana.yaml` file) for the root logger. If we don't have a root appender, then setting the KP log level to `all` won't do anything.

Reply:
Yea, but the config don't have a `rawConfig` property, right?

Reply:
+1, I don't think the `rawConfig.` part is needed. Also it might be a little more clear to reverse the `!` here and have:

```
if (has()) {
  set('logging.root.level, 'all');
} else {
  set('logging.verbose', true);
}
```

Reply:
it does. We [pass](https://github.com/elastic/kibana/blob/master/src/cli/serve/serve.js#L49) the `rawConfig` in to `applyConfigOverrides(rawConfig, opts, extraCliOptions)`

Reply:
The `has` that we are using here isn't the one directly imported from lodash, we're [using `partial` to apply `rawConfig` to it](https://github.com/elastic/kibana/blob/master/src/cli/serve/serve.js#L52), so including `rawConfig` here again is redundant as it's effectively doing the same thing.

Reply:
Thanks everyone! It's been updated.